compendium/ConfigurationAdmin/test/TestConfigAdmin.cpp

```diff
@@ -843,3 +849,101 @@ TEST_F(ConfigAdminTests, testMultipleManagedFactoriesInBundleGetUpdate)
                             1);
                 });
 }
+// <summary>
+/// Used by ConfigAdminTests.testManagedServiceRemoveConfigurationsDeadlock
+/// </summary>
+namespace cppmicroservices {
+namespace test {
+class TestManagedServiceInterface2
+  : public cppmicroservices::service::cm::ManagedService
+{
+public:
+  virtual ~TestManagedServiceInterface2() noexcept = default;
+  void Updated(const cppmicroservices::AnyMap&) override = 0; 
+};
+
+class TestManagedServiceImpl : public TestManagedServiceInterface2
+{
+public:
+  TestManagedServiceImpl(cppmicroservices::Framework& framework)
+    : m_framework(framework)
+   {}
+  virtual ~TestManagedServiceImpl() noexcept = default;
+
+  void Updated(const cppmicroservices::AnyMap& props) override
+  { 
+    // The Updated method is called for both Updated and Removed operations. 
+    // For the Removed operations, the properties are empty.
+    if (props.empty()) {
+      auto ctx = m_framework.GetBundleContext();
+      auto bundles =
+        ctx.GetBundles(PathToLib("TestBundleManagedServiceDeadlock"));
```

Comment:
you want to assert that the # of bundles is 1 and the bundle you got is the correct one (e.g. check the bundle symbolic name) to avoid false positives whereby `GetBundles` returns an empty vector and the test passes because `Stop()` didn't run at all.